src/guides/v2.4/test/integration/annotations/magento-data-fixture.md

```diff
@@ -82,6 +95,137 @@ Test case that uses the above data fixture: [`dev/tests/integration/testsuite/Ma
 
 [`dev/tests/integration/testsuite/Magento/Cms/Controller/PageTest.php`][] demonstrates an example of the `testCreatePageWithSameModuleName()` test method that uses data from the `cmsPageWithSystemRouteFixture()` data fixture.
 
+### Parametrized Fixture class
+
+Define the fixture in a separate Parametrized Fixture class when you think it could be reused in different test cases.
+Fixture class can be provided with a data provider for additional parameters and customizations.
+
+There are two types of data providers:
+-  Inline JSON as data provider
+-  Decoupled fixtures class
+
+#### Inline JSON as data provider
+
+Example:
+
+```php?start_inline=1
+class ProductsList extends \PHPUnit\Framework\TestCase
+{
+    /**
+    * @magentoDataFixture \Magento\Catalog\Fixture\CreateSimpleProduct with:{"sku": "simple1", "price": 5.0}
+    * @magentoDataFixture \Magento\Catalog\Fixture\CreateSimpleProduct with:{"sku": "simple2", "price": 10.0}
+    * @magentoDataFixture \Magento\Catalog\Fixture\CreateSimpleProduct with:{"sku": "simple3", "price": 15.0}
+    */
+    public function testGetProductsCount(): void
+    {
+    }
+}
+```
+
+### A class method as data provider
+
+Example:
+
+```php?start_inline=1
+class ProductsList extends \PHPUnit\Framework\TestCase
+{
+   /**
+   * @magentoDataFixture Magento\Catalog\Fixture\CreateSimpleProduct as:product1
+   * @magentoDataFixture Magento\Catalog\Fixture\CreateSimpleProduct as:product2
+   * @magentoDataFixture Magento\Catalog\Fixture\CreateSimpleProduct as:product3
+   */
+   public function testGetProductsCount(): void
+   {
+   }
+
+   public function getProductsCountFixtureDataProvider(): array
+   {
+      return [
+         [
+             'product1' => [
+                 'sku' => 'simple1'
+             ],
+             'product2' => [
+                 'sku' => 'simple2'
+             ],
+             'product3' => [
+                 'sku' => 'simple3',
+                 'status' => Status::STATUS_DISABLED,
+             ],
+         ]
+      ];
+   }
+}
+```
+
+### Decoupled fixtures class
```

Comment:
I suggest to move this section to a different page under "Best Practices For using Parameterized Fixtures"